distributed/utils.py

```diff
@@ -931,7 +931,9 @@ def ensure_bytes(s):
     >>> ensure_bytes(b'123')
     b'123'
     """
-    if hasattr(s, "encode"):
+    if isinstance(s, bytes):
```

Comment:
I don't believe `bytes(s)` does make a copy if `s` is already `bytes`

Reply:
Hmm...maybe I'm misremembering. In any event we seem to have [similar code in Dask]( https://github.com/dask/dask/blob/79836fe4a52c9cf0e1eb32525792ad6bfed1c62b/dask/utils.py#L772-L773 ).

Reply:
OK, well never mind - this version shouldn't hurt. I wonder why not just call the dask version? I suppose this one can work on memoryviews and bytearrays too.

Reply:
Yeah I was thinking about that as well, but didn't want to go down a rabbit hole here. Am ok pulling this out into a separate PR so we can explore orthogonally. WDYT?

Reply:
I'm saying your version is fine :)

Reply:
FWIW updating the Dask implementation to contain the code from Distributed in PR ( https://github.com/dask/dask/pull/9050 ). Then we should be able to switch to the Dask implementation in Distributed.